packages/lab/src/date-picker/DatePickerOverlayProvider.tsx

```diff
@@ -107,12 +123,26 @@ export const DatePickerOverlayProvider: React.FC<
     }
   }, [open]);
 
-  const setOpen = useCallback((newOpen: boolean) => {
-    if (newOpen) {
-      triggeringElement.current = document.activeElement as HTMLElement;
-    }
-    setOpenState(newOpen);
-  }, []);
+  const setOpen = useCallback(
+    (
+      newOpen: boolean,
+      _event?: Event | undefined,
+      reason?: OpenChangeReason | undefined,
+    ) => {
+      if (newOpen) {
+        triggeringElement.current = document.activeElement as HTMLElement;
+      }
+      setOpenState(newOpen);
+      onOpen?.(newOpen);
+      if (
+        reason === "escape-key" ||
+        (reason === "outside-press" && onDismissCallback.current)
+      ) {
+        onDismissCallback?.current?.();
+      }
+    },
+    [onOpen],
+  );
 
   const floatingUIResult = useFloatingUI({
```

Comment:
Placement is hard coded here, can we expose things like `placement` from `Tooltip`? SO 69481

Reply:
good shout, follow-up PR though... Ideal target state would be an equivalent, common component 
For now, will make this component support placement

@joshwooding previously discussed